### DISCUSSION TITLE
Share implementation of possibleTypes

### DISCUSSION THREAD

**Discussion by knaeckeKami** [heftapp/graphql_codegen#153] [General]
Hi! 

I am picking up maintainence of gql-dart and ferry.
I saw that since normalize 0.6.0, the behavior around fragments was improved when we have a possibleTypes map.

I saw that you have implemented the generation of such a map.

Do you think it makes sense to extract this logic so we can use it for ferry as well?

Maybe in gql_code_builder ? You depend on that so we could share the implementation

**Comment by budde377** [comment]
I think generating a list of possible types from a schema would be an excellent extension to `normalized` as that's where we're using the map. 

Whether we're going to replace this implementation I'm not sure. The algorithm seems relatively stable and straightforward, so I'm having a hard time seeing what the benefits are. Of course, we'll be sharing the logic, but the way we handle schemas in the two libraries are probably different and so is the way we write the output. 

But I have absolutely no problem with you extracting the logic and sharing it with ferry.

**Reply by knaeckeKami** [reply]
Alright. I'll look into adding this to gql-dart. Yeah, the algorithm seems simple enough that sharing it might complicate things more.
